services/gateway/shared/registerHttpApi.js

```diff
@@ -64,137 +94,143 @@ const configureApi = (apiNames, apiPrefix, registeredModuleNames) => {
 		return { ...acc, [key]: method };
 	}, {});
 
-	const whitelist = Object.keys(methods).reduce((acc, key) => [
-		...acc, methods[key].source.method,
-	], []);
-
-	const aliases = Object.keys(methods).reduce((acc, key) => ({
-		...acc, [`${getMethodName(methods[key])} ${transformPath(methods[key].swaggerApiPath)}`]: methods[key].source.method,
-	}), {});
-
-	const methodPaths = Object.keys(methods).reduce((acc, key) => ({
-		...acc, [`${getMethodName(methods[key])} ${transformPath(methods[key].swaggerApiPath)}`]: methods[key],
-	}), {});
-
-	const meta = {
-		apiPrefix,
-		routes: Object.keys(methods).map(m => ({
-			path: methods[m].swaggerApiPath,
-			params: Object.keys(methods[m].params || {}),
-			response: {
-				...methods[m].envelope,
-				...methods[m].source.definition,
-			},
-		})),
-	};
-
-	apiMeta.push(meta);
-
-	return { aliases, whitelist, methodPaths };
+	return methods;
 };
 
-const registerApi = (apiNames, config, registeredModuleNames) => {
-	const { aliases, whitelist, methodPaths } = configureApi(
-		apiNames,
-		config.path,
-		registeredModuleNames,
-	);
-
-	return {
-		...config,
-
-		whitelist: [
-			...config.whitelist,
-			...whitelist,
-		],
-
-		aliases: {
-			...config.aliases,
-			...aliases,
-		},
+const getAPIConfig = (configPath, config, aliases, whitelist, methodPaths, etag) => ({
+	...config,
+
+	path: configPath,
+
+	whitelist: [
+		...config.whitelist,
+		...whitelist,
+	],
+
+	aliases: {
+		...config.aliases,
+		...aliases,
+	},
+
+	etag: (etag === undefined || etag === 'strong') ? 'strong' : false,
```

Comment:
Please verify the suggestion.
```suggestion
	etag: (typeof etag === function) ? etag() : etag,
```